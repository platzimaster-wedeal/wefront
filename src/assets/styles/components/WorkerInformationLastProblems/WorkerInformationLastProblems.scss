@import '../../partials/colors.scss';
@import '../../partials/sizes.scss';
@import '../../partials/fonts.scss';
@import '../../FontSize.scss';
@import '../../Breakpoints.scss';

.worker-information-last-problems {
  position: relative;
  max-width: 320px;
  width: auto;
  padding: $space-size-big;

  @include media(tablet){
    max-width: 50%;
  }
  
  &__title {
    margin: $space-size-extra-small 0;
    font-weight: 400;
    @include font-size($font-size-normal);

    @include media(tablet) {
      @include font-size($font-size-medium);
    }
  }

  &__list li{
    
    @include font-size($font-size-small);

    @include media(tablet) {
      @include font-size($font-size-normal);
    }  
  }
}