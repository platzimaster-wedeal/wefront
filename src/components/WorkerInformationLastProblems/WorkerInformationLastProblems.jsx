import React from "react";

// Styles
import "../../assets/styles/components/WorkerInformationLastProblems/WorkerInformationLastProblems.scss";

// Components
import ProblemPrevisualization from "../ProblemPrevisualization/ProblemPrevisualization";

const WorkerInformationLastProblems = ({ problems }) => {
 return (
  <div className="worker-information-last-problems">
   <strong className="worker-information-last-problems__title">
    Last Problems Solved
   </strong>
   <ul className="worker-information-last-problems__list">
    {problems && problems.length > 0 ? (
     problems.map((problem, id) => (
      <li key={id}>
       {" "}
       <ProblemPrevisualization
        state={problem.state}
        idProblem={problem.id}
       />{" "}
      </li>
     ))
    ) : (
     <span>This user doesn't have solve any problem yet.</span>
    )}
   </ul>
  </div>
 );
};

export default WorkerInformationLastProblems;
