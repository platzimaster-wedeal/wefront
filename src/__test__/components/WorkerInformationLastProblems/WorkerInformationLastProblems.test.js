import React from "react";
import { mount } from "enzyme";
import { create } from "react-test-renderer";
import WorkerInformationLastProblems from "../../../components/WorkerInformationLastProblems/WorkerInformationLastProblems";

describe("Test for the component WorkerInformationLastProblems :D", () => {
 const WorkerInformationLastProblemsMounted = mount(
  <WorkerInformationLastProblems problems={["new problem"]} />
 );
 const WorkerInformationLastProblemsCreated = create(
  <WorkerInformationLastProblems />
 );

 describe("Visual tests for the component", () => {
  test("should the comopnent have the same tree", () => {
   expect(WorkerInformationLastProblemsCreated.toJSON()).toMatchSnapshot();
  });
 });

 describe("Functional tests for the component", () => {
  test("Should the component render one element", () => {
   expect(WorkerInformationLastProblemsMounted.length).toEqual(1);
  });
  test("Should the title of the component be a string", () => {
   expect(
    typeof WorkerInformationLastProblemsMounted.find(
     ".worker-information-last-problems__title"
    ).text()
   ).toEqual("string");
  });
  test("Should the title of the component be equal to Last Problems Solved", () => {
   expect(
    WorkerInformationLastProblemsMounted.find(
     ".worker-information-last-problems__title"
    ).text()
   ).toEqual("Last Problems Solved");
  });

  test("Should the description of null information be a string", () => {
   expect(
    typeof mount(<WorkerInformationLastProblems />)
     .find(".worker-information-last-problems__list span")
     .text()
   ).toEqual("string");
  });
  test("Should the description of the component be equal to This user doesn't have certificates upload", () => {
   expect(
    mount(<WorkerInformationLastProblems />)
     .find(".worker-information-last-problems__list span")
     .text()
   ).toEqual("This user doesn't have solve any problem yet.");
  });
 });
});
