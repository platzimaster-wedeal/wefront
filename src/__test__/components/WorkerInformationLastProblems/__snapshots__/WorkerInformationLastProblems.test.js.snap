// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for the component WorkerInformationLastProblems :D Visual tests for the component should the comopnent have the same tree 1`] = `
<div
  className="worker-information-last-problems"
>
  <strong
    className="worker-information-last-problems__title"
  >
    Last Problems Solved
  </strong>
  <ul
    className="worker-information-last-problems__list"
  >
    <span>
      This user doesn't have solve any problem yet.
    </span>
  </ul>
</div>
`;
